### ISSUE TITLE
Mangled component names in production build

### ISSUE THREAD

**Issue by pascalduez** [storybooks/storybook#5546]
Hi,

upgrading from `@storybook/react` 3 to 4 brings back mangled component names in the production build. So in our styleguide which is using  [`react-element-to-jsx-string`](https://github.com/algolia/react-element-to-jsx-string/blob/333591a19420cb52f93b7c638bcaa2ca10bf2d11/src/parser/parseReactElement.js#L12-L17) can't find function names.  

This resulting in our code samples looking like:
```js
<No Display Name />
```

This was fixed in version 3 with: #2705 

So basically it would only boils down to adding back that config to terser, but asking for feedback here first.
